squape/video.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Cyber Alpaca
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import os
+from contextlib import contextmanager
+
+import squishinfo
+
+import test
+
+
+def _failures_results_count() -> int:
+    """
+    This function returns the total number of failures in test results by adding up
+    all the counts for errors, fatals, xpasses and fails.
+
+    Returns:
+        int: The total number of failures in test results.
+    """
+    return (
+        test.resultCount("errors")
+        + test.resultCount("fatals")
+        + test.resultCount("xpasses")
+        + test.resultCount("fails")
+    )
+
+
+def _videos_set() -> set:
+    """
+    This function returns a set of existing video filenames in video directory
+    RESULT_DIR/TEST_SUITE_NAME/TEST_CASE_NAME/attachments.
+
+    Returns:
+        set: The set of filenames with mp4 extension
+    """
+    video_dir = os.path.join(
+        squishinfo.resultDir,
+        squishinfo.testSuiteName,
+        squishinfo.testCaseName,
+        "attachments",
+    )
+
+    videos_list = set()
+
+    if not os.path.exists(video_dir):
+        return videos_list
+
+    for file in os.listdir(video_dir):
+        if file.endswith(".mp4"):
+            videos_list.add(file)
+    return videos_list
+
+
+def _remove_videos(videos: set) -> None:
+    """
+    Remove videos from the attachments directory.
+
+    Args:
+        videos (set): A set of video names to remove.
+    """
+    video_dir = os.path.join(
+        squishinfo.resultDir,
+        squishinfo.testSuiteName,
+        squishinfo.testCaseName,
+        "attachments",
+    )
+    for video_name in videos:
+        test.log(f"Remove video: {video_name}")
+        file_name = os.path.join(video_dir, video_name)
+        os.remove(file_name)
```

Comment:
what if the file doesn't exist?

Reply:
The comment is outdated as after the latest changes videos are not removed.